Adds Gradle build that delegates to existing Ant build

NicoPiel opened the pull request:
Fixes #52

Test with `./gradlew clean build -PdisableTests=true -PdisableSigning=true `

Adds an initial Gradle build system that delegates to the existing Ant build to enable gradual migration and tooling improvements.

Highlights:
- Introduces a root Gradle configuration that imports and delegates to the project's Ant build targets, avoiding immediate rewrite of existing build logic.
- Adds the Gradle wrapper and supporting files (wrapper properties, jar, and scripts) so the repository can be built consistently without requiring a local Gradle install.
- Adds per-subproject Gradle stub files to make the repository a multi-project Gradle layout and allow incremental conversion of subprojects.
- Adds Gradle metadata/configuration (gradle.properties, dependency platform TOML)
- Adds .gitattributes and .gitignore tweaks and normalizes an IDE classpath file to remove trailing encoding differences.
- Configures Ant-to-Gradle interoperability (ant taskdefs and properties) and maps Ant targets to avoid conflicts with Gradle built-in tasks.

Why:
- Allows us to start using Gradle tooling and the wrapper for builds, CI, and IDEs while preserving the working Ant build.
- Supports incremental migration of subprojects to Gradle without disrupting existing build behavior.

Notes / Next steps:
- Subprojects remain backed by the Ant build; plan incremental conversion of individual modules to native Gradle builds.
- Verify CI pipelines and developer SDK setups pick up the wrapper and SDKMAN gradle setting.

NicoPiel committed:
Add root settings and gradle to SDKMAN

Signed-off-by: Nico Piel <nico.piel@hotmail.de>

NicoPiel committed:
Gradle init and build files for all subprojects

Signed-off-by: Nico Piel <nico.piel@hotmail.de>

NicoPiel committed:
Add Gradle build that delegates to existing Ant build

Adds a top-level Gradle build that imports the existing Ant build to enable a gradual migration to Gradle while still using Ant for current tasks. Configures Ant to run JUnit (including ant-junit4) and exposes select Gradle properties to Ant; renames conflicting Ant targets to avoid collisions and makes the Gradle build task delegate to the Ant build target. Also adds placeholder Gradle files for subprojects and bumps the configured Gradle SDK version.

Co-authored-by: Mitch Gaffigan <mitch.gaffigan@comcast.net>
Signed-off-by: Nico Piel <nico.piel@hotmail.de>

NicoPiel committed:
Update docs link

Signed-off-by: Nico Piel <nico.piel@hotmail.de>

NicoPiel committed:
Upgrade Gradle Wrapper to 9.2.1

Signed-off-by: Nico Piel <nico.piel@hotmail.de>

mgaffigan commented on gradle/wrapper/gradle-wrapper.properties:
```diff
@@ -0,0 +1,7 @@
+distributionBase=GRADLE_USER_HOME
+distributionPath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-9.2.1-all.zip
```
Do we want `-all.zip`?  This is apparently intended for offline use.  Default initialize gradle script uses `-bin.zip`.

NicoPiel replied:
This is what `gradle init` generated. I can use the bin version, though.

mgaffigan reviewed (changes requested):
Other issues to address
- CI build scripts need updated (.github/workflow/...)
- Readme/Contributor documents need updated
- ant mirth-build build target needs to be updated to block build outside of gradle with migration guidance in error message

NicoPiel commented on gradle/libs.versions.toml:
```diff
@@ -0,0 +1,8 @@
+guava = "33.4.5-jre"
+
+[libraries]
+guava = { module = "com.google.guava:guava", version.ref = "guava" }
```
It may be something Gradle uses internally.

tonygermano commented on .sdkmanrc:
```diff
@@ -2,3 +2,4 @@
 # Add key=value pairs of SDKs to use below
 java=17.0.17.fx-zulu
 ant=1.10.14
+gradle=9.2.1
```
I don't think this change is necessary. The gradle version is handled by the gradle wrapper.

tonygermano commented on settings.gradle.kts:
```diff
@@ -0,0 +1,19 @@
+    id("org.gradle.toolchains.foojay-resolver-convention") version "0.10.0"
+}
+
+rootProject.name = "OIE"
```
agree with the lowercase. I think eventually the project name gets included in the jar filenames, so I'd say "oie" just for brevity.

tonygermano commented on build.gradle.kts:
```diff
@@ -0,0 +1,71 @@
+
+project.findProperty("disableTests")?.let {
+    ant.properties["disableTests"] = it.toString()
+}
```
https://github.com/OpenIntegrationEngine/engine/commit/9adf3f33cf807f2cedafb227eb112c0faf10b0c6

This commit added `disableSigning` as well as `cert` and `keystore_property_file`

NicoPiel committed:
Normalize Gradle project name and classpath formatting

Changes the root project name to a lowercase value to avoid case-related build/tooling issues (improves compatibility with Gradle/Ant and CI environments).

Signed-off-by: Nico Piel <nico.piel@hotmail.de>

NicoPiel committed:
Remove gradle from SDKMAN env

Signed-off-by: Nico Piel <nico.piel@hotmail.de>

NicoPiel committed:
Use Gradle wrapper in CI and configure Gradle root to bridge to Ant build

CI workflow now invokes the Gradle wrapper (./gradlew build) for signed and unsigned builds instead of calling Ant directly. The root Gradle build was extended to set project group/version, provide repositories for subprojects, and pass the coverage and disableTests properties through to the imported Ant build. Added version and archiveEntryDate to gradle.properties and updated the settings docs link.

This lets the top-level Gradle build act as a thin wrapper around the existing Ant build, enabling gradual migration to Gradle while keeping CI changes minimal.

Signed-off-by: Nico Piel <nico.piel@hotmail.de>
